_example/fileconfig.go

```diff
@@ -21,11 +22,19 @@ type ExampleFileConfig struct {
 func (e *ExampleFileConfig) Validate(component diag.Component) diag.Diagnostics {
 	var diags diag.Diagnostics
 	if e.MyApp.Port < 0 || e.MyApp.Port > 65535 {
```

Comment:
if this was more 'real world' the port validation and logging could be done via a function.